stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarView.kt

```diff
@@ -114,30 +166,89 @@ public class AvatarView : AppCompatImageView {
         }
     }
 
-    private fun drawBorder(canvas: Canvas) {
-        if (avatarStyle.avatarBorderWidth != 0) {
+    private fun avatarShape(style: AvatarStyle): StreamImageLoader.ImageTransformation {
```

Comment:
Can we use the global `avatarStyle` property instead of passing the parameter for this and `drawColor` functions?

Reply:
We can, but then we lose [referential transparency.](https://www.sitepoint.com/what-is-referential-transparency/) . Right not, this functional is a method that could be used in any place that can pass `AvatarStyle` as an argument, but we if use the class property, this method gets coupled to this class. 

Using parameters instead makes the code easier to reason about (in my option) because the state that the method has to deal with is only the state of the parameters passed to it instead of the state of the whole class

Reply:
That's a great point! Then should we refactor the `drawOnlineStatus` to receive the `avatarStyle ` for consistency?